test/test_uri_parser.py

```diff
@@ -553,6 +554,72 @@ def test_port_with_whitespace(self):
         with self.assertRaisesRegex(ValueError, r"Port contains whitespace character: '\\n'"):
             parse_uri("mongodb://localhost:27\n017")
 
+    def test_allow_srv_hosts_with_fewer_than_three_dot_separated_parts(self):
+        with patch("dns.resolver.resolve"):
+            parse_uri("mongodb+srv://localhost/")
+            parse_uri("mongodb+srv://mongo.local/")
+
+    def test_error_when_return_address_does_not_end_with_srv_domain(self):
```

Comment:
Could you split this into 3 separate tests so we can correspond them with the spec?